### PR TITLE
feat(hints): add NewHint#13

### PR DESCRIPTION
**Pull request by MegaRedHand**
## Description

This PR adds the missing hint _NewHint#13_, now known as `QUAD_BIT`. I placed the implementation in the `secp::ec_utils` module because it's used in _[point multiplication via Shamir's trick + Windowed method](https://github.com/myBraavos/efficient-secp256r1/blob/73cca4d53730cb8b2dcf34e36c7b8f34b96b3230/src/secp256r1/ec_mulmuladd.cairo#L87)_.

### CODE REVIEW

File: src/hint_processor/builtin_hint_processor/secp/ec_utils.rs [hunk 1]
```diff
@@ -809,4 +855,51 @@ mod tests {
...
+
+    #[test]
+    fn run_quad_bit_with_max_m_ok() {
+        let hint_code = "ids.quad_bit = (\n    8 * ((ids.scalar_v >> ids.m) & 1)\n    + 4 * ((ids.scalar_u >> ids.m) & 1)\n    + 2 * ((ids.scalar_v >> (ids.m - 1)) & 1)\n    + ((ids.scalar_u >> (ids.m - 1)) & 1)\n)";
+        let mut vm = vm_with_range_check!();
+
+        let scalar_u = 89712;
+        let scalar_v = 1478396;
+        // Value is so high the result will always be zero
+        let m = i128::MAX;
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/secp/ec_utils.rs:890]: This should be true for anything `>= 253`. That's how many bits the Cairo prime takes.

**Review by Oppen — APPROVED** [review]
LGTM

File: src/hint_processor/builtin_hint_processor/secp/ec_utils.rs (hunk 1 above)
Comment (MegaRedHand) [reply src/hint_processor/builtin_hint_processor/secp/ec_utils.rs:893]: 😮 I didn't think of that! I changed the check inside `quad_bit`. Thanks for the help!

File: src/hint_processor/builtin_hint_processor/secp/ec_utils.rs [hunk 2]
```diff
@@ -298,8 +298,11 @@ pub fn quad_bit(
     let scalar_u = scalar_u_cow.as_ref();
 
     // If m is too high the shift result will always be zero
-    let Some(m) = m_cow.as_ref().to_u32() else {
-        return insert_value_from_var_name("quad_bit", 0, vm, ids_data, ap_tracking);
+    let m = match m_cow.as_ref().to_u32() {
+        Some(m) if m <= 252 => m,
+        None | Some(_) => {
+            return insert_value_from_var_name("quad_bit", 0, vm, ids_data, ap_tracking)
+        }
```
Comment (Oppen) [inline src/hint_processor/builtin_hint_processor/secp/ec_utils.rs:305]: suggestion: dirty trick
```suggestion
    let m = match m_cow.as_ref().to_u32().unwrap_or(253);
    if m >= 253 {
            return insert_value_from_var_name("quad_bit", 0, vm, ids_data, ap_tracking);
    }
```

**Review by Oppen — APPROVED** [review]

**Review by Oppen — APPROVED** [review]

**Review by fmoletta — APPROVED** [review]

**Pull request merged**